### PR TITLE
fix: MD003/heading-style/header-style

### PR DESCRIPTION
**Pull request by nschonni**
Mostly from C# at the end of the titles.
This can trigger parsing confusion because that is an alternate header style

### CODE REVIEW

**Comment by PRMerger7** [comment]
@nschonni : Thanks for your contribution! The author, @mikejo5000, has been notified to review your proposed change.

**Comment by ktoliver** [comment]
@Mikejo5000 @gewarren Could you review the proposed changes? Thanks.

**Comment by Mikejo5000** [comment]
@ghogen Gordon, what do you think of this?

File: docs/code-quality/C26414.md [hunk 1]
```diff
@@ -27,9 +27,10 @@ Smart pointers are convenient for dynamic resource management, but they are not
   - public destructor which is neither deleted nor defaulted. This includes destructors which are explicitly defined empty.
   - The type Microsoft::WRL::ComPtr behaves as a shared pointer, but it is often used in quite specific scenarios which are affected by the COM lifetime management. To avoid excessive noise this type is filtered out.
   - This check looks for explicit local allocations assigned to smart pointers to identify if scoped variables could word as an alternative. In addition to direct calls to operator new, special functions like std::make_unique and std::make_shared are also interpreted as direct allocations.
-    -
-    ## Example
-    dynamic buffer
+
+## Example
```
Comment (nschonni) [inline docs/code-quality/C26414.md]: This one is also covered by #2765

**Comment by ghogen** [comment]
I'm not convinced this is necessary or desirable.

**Comment by nschonni** [comment]
Parsers can confuse these headings with the Closed ATX style https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md003

**Comment by Mikejo5000** [comment]
I wouldn't want to do this unless the DotNet people were in favor of it. They will have a lot of C# references in their headings.

**Comment by nschonni** [comment]
@Mikejo5000 I'll take a look at there repo too. Note that this only is an "issue" when the final character is a hash. Just having a hash in a title isn't a problem.

**Comment by Mikejo5000** [comment]
@mairaw FYI

**Comment by nschonni** [comment]
@Mikejo5000 https://github.com/dotnet/docs/pull/10714 landed on the DotNet docs. @mairaw pointed out that there is a bug in the TOC generation right now, that means any of the H2s can't be escaped right now

File: docs/ide/managing-application-settings-dotnet.md [hunk 2]
```diff
@@ -78,7 +78,7 @@ If any user-scoped settings are changed during run time, for example in testing
 
 We strongly recommend that you use the `My.Settings` object and the default *.settings* file to access settings. This is because you can use the **Settings Designer** to assign properties to settings, and, additionally, user settings are automatically saved before application shutdown. However, your Visual Basic application can access settings directly. In that case you have to access the `MySettings` class and use a custom *.settings* file in the root of the project. You must save the user settings before ending the application, as you would do for a C# application; this is described in the following section.
 
-## Access or change application settings at run time in C# #
+## Access or change application settings at run time in C\#
```
Comment (nschonni) [inline docs/ide/managing-application-settings-dotnet.md]: This will need to be reverted pending the TOC bug

**Comment by gewarren** [comment]
I've moved these commits to the private repo for staging verification.

**Pull request closed without merging**